edx_sga/sga.py

```diff
@@ -652,6 +691,67 @@ def remove_grade(self, request, suffix=''):
         )
         return Response(json_body=self.staff_grading_data())
 
+    @XBlock.handler
+    def prepare_download_submissions(self, request, suffix=''):  # pylint: disable=unused-argument
+        """
+        Runs a async task that collects submissions in background and zip them.
+        """
+        require(self.is_course_staff())
+        user = get_user_by_id(getattr(self.xmodule_runtime, 'user_id', None))
+
+        if is_zip_file_available(user, self.block_course_id, unicode(self.block_id), self.location):
+            assignments = sort_assignment_by_date(self.staff_grading_data()['assignments'])
```

Comment:
I'm fine with either of these approaches, but get_all_submissions and/or querying the submissions model directly may raises some red flags with edX when it comes time to merge upstream. It might be worth asking or looking for a more specific API that does what we need.

Reply:
@pdpinch we are already querying these models in current SGA, i am just reusing that code

Reply:
@pdpinch On top of that, the submissions API only has 5 methods or so. I'm confident that we're not missing a method that does what we need